src/pubkey.h

```diff
@@ -136,22 +136,22 @@ class CPubKey
     {
         unsigned int len = size();
         ::WriteCompactSize(s, len);
-        s.write((char*)vch, len);
+        s.write({BytePtr(vch), len});
     }
     template <typename Stream>
     void Unserialize(Stream& s)
     {
         unsigned int len = ::ReadCompactSize(s);
         if (len <= SIZE) {
-            s.read((char*)vch, len);
+            s.read({BytePtr(vch), len});
             if (len != size()) {
                 Invalidate();
             }
         } else {
             // invalid pubkey, skip available data
```

Comment:
I think this could just be `s.ignore(len);`.

Reply:
Good idea. Done in https://github.com/bitcoin/bitcoin/pull/23451